172534313 avance de la historia

Adrian-de-Lucas-Gomez opened the pull request:
Añadido la escena entre partidas en la que vemos como avanzan las rondas y como van quedando los jugadores. La nueva escena se llama MidGame y se puede acceder a ella pulsando el Select en tu mando estando en uno de los mapas.

Esta preparado para distinto número de jugadores, adaptando su colocación en pantalla y también para diferente número de rondas. Para indicar cuantas partidas ha ganado un jugador se sabe por cuanto ha avanzado por la pantalla.

La escena comienza con los cohetes parados y avanza el que ha ganado la última ronda. Tras esto aparece el pantalla el botón de continuar y se hace una transición tope guapa y se pasaría al siguiente mapa (El que decide la clase superior que controla las partidas).

Hay parámetros que de momento están puestos a pelo pero estos nos los dará la clase superior. Al meterte una segunda vez al estado verás que los cohetes están en otra posición pero esto es normal ya que simula que se han actualizado las puntuaciones.

No se asusten si nada mas entrar al estado le dais al botón a pasará directamente a la animación para salir del estado (para pasarlo rápido como dijo Kleon).

mmur07 committed:
*screams porque se rompe el viewer*

Adrian-de-Lucas-Gomez committed:
Imagen del cohete y estacion espacial

Aun estan en progreso pero es para que tengamos imagenes ya asociadas a las entidades.

mmur07 committed:
ya hay animación básica de los cohetes

mmur07 committed:
tenemos estación espacial y no explota al entrar

mmur07 committed:
subidos cambios para espiar otras ramas

mmur07 committed:
Merge branch 'master' into 172534313-AvanceDeLaHistoria

mmur07 committed:
arreglos del merge

Adrian-de-Lucas-Gomez committed:
Cohetes y fondo de estrellitas

Ya tenemos los cohetes haciendo la animación y tenemos el fondico con las estrellas

mmur07 committed:
commit para mergear master

mmur07 committed:
Merge branch 'master' into 172534313-AvanceDeLaHistoria

mmur07 committed:
los cohetes ya tienen la animación bien puesta

el codigo es omegaultrafeodecojones

mmur07 committed:
la transición entre escenas está terminada. No peta asi que nice

Adrian-de-Lucas-Gomez committed:
MidGame esta casi listo

Para acceder a el ahora se debe pulsar el  boton de home en el Pro Controller de la Switch, no se a cual equivale en uno normal. Si se necesita cambiar esta en el InputHandler de PlayState

Adrian-de-Lucas-Gomez committed:
Paso de parámetros a constantes

Se han movido muchas variables de las configurables a las constantes. Si se quiere se puede saltar el movimiento de las naves para pasar mas rápido el estado de Midgame.

Adrian-de-Lucas-Gomez committed:
Marcadores para los puntos

Aun no tienen un sprite propio pero ya se colocan en función de cuantas rondas se vayan a jugar.

Adrian-de-Lucas-Gomez committed:
Prubas en pantalla 1080p

Como mi portatil tiene una resolucion inferior he tenido que robar una tele para ver si las cosas estaban bien colocadas y así aprivechar todo el espacio posible.

Adrian-de-Lucas-Gomez committed:
Ajustes y sprites

Se han puesto iconos personalizados para los marcadores de los puntos, se ha oscurecido la estación espacial y se ha ajustado el punto de origen de los cohetes.

Adrian-de-Lucas-Gomez committed:
Ajustes para el debug del MigGame

Se ha cambiado el acceso al estado. Ahora se accede pulsando hacia arriba en la cruceta.

Adrian-de-Lucas-Gomez committed:
Cambio de boton y control remoto proporcionado

Ahora accedemos al modo MidGame con el select porque sino Adri me pega.
Ahora el mando de la tele es alargado, como debe ser

Adrian-de-Lucas-Gomez committed:
Merge branch 'master' into 172534313-AvanceDeLaHistoria

EstebenR committed:
Update PCE_Project.vcxproj.filters

EstebenR reviewed (changes requested):
molto bonito, but

Si viewer tiene un componente nFrames_ animatedViewer no debería redefinir otro con el mismo nombre y menos luego en el init cambiar el primero. ¿Por que existen dos?

El texto dice pulsa A para continuar, intuyendo que es la A del mando no tiene soporte para otros controles, igual mejor poner impulso o atacar o agarrar o lo que sea que sea mas universal (Resources.cpp linea 123)

el nido de else if en MidGameState.cpp quedarian mas limpios con un switch

Adrian-de-Lucas-Gomez committed:
Cambios por limpieza y estructura

Ya se han solucionado los problemas que encontró Esteban. Ahora con su permiso me voy a sobar muy fuerte.

EstebenR reviewed (changes requested):
El codigo sigue pidiendo que le des a la A, no al impulso xd

Adrian-de-Lucas-Gomez committed:
Cambios para pasar el estado

Ahora no te pide ninguna tecla espacífica, esto lo hace mas cómodo y evita tener que disponer de referencias a los binders de los jugadores. Funciona en teclado y en mando pulsando lo que quieras.
Tambien he cambiado el mensaje

Adrian-de-Lucas-Gomez committed:
Merge branch 'master' into 172534313-AvanceDeLaHistoria

EstebenR reviewed (approved):
Good save

eljorgemo commented on src/SourceFiles/MidGameState.h:
```diff
@@ -0,0 +1,62 @@
+		numPlayers_ = numPlayers; 
+		roundWinner_ = roundWinner;
+	};
+	~MidGameState();
```
```suggestion
	virtual ~MidGameState();
```

eljorgemo committed:
destructora de MidGameState virtual

JuanmaSalva reviewed (approved):
to wapa la animacion
